File: 3-BIF#2024-2025/CORSI-VARIE/CorsoCyberSecurity/codiceCesare.cpp
/*
Dati in input un testo e un numero intero N minore di 26,
restituire in uscita il testo crittografato con un Codice
di Cesare che sposti ogni lettera di N lettere.
*/

#include <iostream>
#include <string>
using namespace std;

// Funzione per crittografare il testo usando il Codice di Cesare
string caesarCipher(string text, int shift)
{
    string result = "";
    // Itera su ogni carattere del testo
    for (int i = 0; i < text.length(); i++)
    {
        // Controlla se il carattere è una lettera
        if (isalpha(text[i]))
        {
            // Determina
            {
                result += text[i];
            }
        }
        return result;
    }

    int main()
    {
        int N = 0;
        cout << "Il valore di N deve essere minore di 26..." << endl;
        cout << "Dammi il valore di N, da tastiera: " << endl;
        cin >> N;
        while (N >= 26)
        {
            cout << "Il valore di N è MAGGIORE o UGUALE a 26..." << endl;
            cout << "...deve essere MINORE..." << endl;
            cout << "Riprovare: ";
            cin >> N;
        }

        cin.ignore(); // to ignore the newline character left in the buffer
        string text;
        cout << "Inserisci il testo da crittografare: ";
        getline(cin, text);

        string encryptedText = caesarCipher(text, N);
        cout << "Testo crittografato: " << encryptedText << endl;

        return 0;
    }

File: 3-BIF#2024-2025/CORSI-VARIE/CorsoCyberSecurity/codiceCesare.py
import os

# Funzione per cambiare il caso delle lettere nel file
def change_case(file_path):
    with open(file_path, 'r') as file:
        content = file.read()

    new_content = ""
    for char in content:
        if char.islower():
            new_content += char.upper()
        elif char.isupper():
            new_content += char.lower()
        else:
            new_content += char

    with open(file_path, 'w') as file:
        file.write(new_content)

# Funzione per riformattare il testo rimuovendo spazi extra
def reformat_text(file_path):
    with open(file_path, 'r') as file:
        content = file.read()

    new_content = ' '.join(content.split())

    with open(file_path, 'w') as file:
        file.write(new_content)

# Funzione per crittografare il testo usando il Codice di Cesare
def caesar_cipher(file_path, shift):
    with open(file_path, 'r') as file:
        content = file.read()

    new_content = ""
    for char in content:
        if char.isalpha():
            shift_amount = shift % 26
            if char.islower():
                new_content += chr((ord(char) - ord('a') + shift_amount) % 26 + ord('a'))
            elif char.isupper():
                new_content += chr((ord(char) - ord('A') + shift_amount) % 26 + ord('A'))
        else:
            new_content += char

    with open(file_path, 'w') as file:
        file.write(new_content)

# Funzione principale che gestisce il menu e le scelte dell'utente
def main():
    file_path = input("Inserisci il percorso del file (.cpp, .txt, .py): ")

    if not os.path.isfile(file_path):
        print("Il file non esiste.")
        return

    print("Menu:")
    print("1. Cambiare da MAIUSCOLE a minuscole e viceversa")
    print("2. Riformattare il testo")
    print("3. Crittografare il testo con il Codice di Cesare")

    choice = input("Scegli un'opzione (1, 2 o 3): ")

    if choice == '1':
        change_case(file_path)
        print("Il caso del testo è stato cambiato con successo.")
    elif choice == '2':
        reformat_text(file_path)
        print("Il testo è stato riformattato con successo.")
    elif choice == '3':
        shift = int(input("Inserisci il numero di posizioni per il Codice di Cesare (N): "))
        if 0 < shift < 26:
            caesar_cipher(file_path, shift)
            print("Il testo è stato crittografato con successo.")
        else:
            print("Il numero di posizioni deve essere compreso tra 1 e 25.")
    else:
        print("Scelta non valida.")

if __name__ == "__main__":
    main()
